File: flutter_module/lib/features/bluetooth/domain/entities/bluetooth_device.dart
import 'package:equatable/equatable.dart';

class BluetoothDevice extends Equatable {
  final String? name;
  final String address;
  final int rssi;
  final bool isConnected;
  final DeviceType deviceType;
  final List<String> services;

  const BluetoothDevice({
    this.name,
    required this.address,
    required this.rssi,
    this.isConnected = false,
    this.deviceType = DeviceType.unknown,
    this.services = const [],
  });

  @override
  List<Object?> get props => [name, address, rssi, isConnected, deviceType, services];
}

enum DeviceType { classic, ble, dual, unknown }

enum ConnectionState { disconnected, connecting, connected, disconnecting, error }

class ConnectionStatus extends Equatable {
  final ConnectionState state;
  final BluetoothDevice? device;
  final String? message;

  const ConnectionStatus({
    required this.state,
    this.device,
    this.message,
  });

  @override
  List<Object?> get props => [state, device, message];
}

File: app/src/main/java/com/example/androidapp/data/model/BluetoothDevice.kt
package com.example.androidapp.data.model

data class BluetoothDevice(
    val name: String?,
    val address: String,
    val rssi: Int,
    val isConnected: Boolean = false,
    val bondState: Int = 0,
    val deviceType: DeviceType = DeviceType.UNKNOWN,
    val services: List<String> = emptyList()
)

enum class DeviceType {
    CLASSIC,
    BLE,
    DUAL,
    UNKNOWN
}

data class ConnectionStatus(
    val state: ConnectionState,
    val device: BluetoothDevice?,
    val message: String? = null
)

enum class ConnectionState {
    DISCONNECTED,
    CONNECTING,
    CONNECTED,
    DISCONNECTING,
    ERROR
}